dialogy/plugins/text/list_entity_plugin/__init__.py

```diff
@@ -25,7 +28,8 @@
 Label = str
 Span = Tuple[int, int]
 Value = str
-MatchType = List[Tuple[Text, Label, Value, Span]]
+Score = float
+MatchType = List[Tuple[Text, Label, Value, Span, Score]]  # adding score for each entity
 
 
 class ListEntityPlugin(EntityExtractor):
```

Comment:
@ltbringer and @AchintyaX , why are we merging list search and previous regex system?

Let's keep them separate. Check out Core ML's documentation on these.

Reply:
Hey @lepisma there was an urgent requirement for this parser is some clients like Jarvis and Vodaphone, so I thought it would be faster if we add it  in `ListEntityPlugin` .  I have gone through the doc for the List Search problem and the proposed structure of the plugin, I would also refer to the doc shared by @ltbringer for writing my own plugin for updates in the algorithm for List Search